iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/pipe/coordinator/task/PipeTaskCoordinator.java

```diff
@@ -73,6 +73,17 @@ public AtomicReference<PipeTaskInfo> tryLock() {
     return null;
   }
 
+  /**
+   * Lock the pipe task coordinator.
+   *
+   * @return the {@link PipeTaskInfo} holder, which can be used to get the {@link PipeTaskInfo}.
+   *     Wait until lock is acquired
+   */
+  public AtomicReference<PipeTaskInfo> lock() {
+    pipeTaskCoordinatorLock.lock();
+    return new AtomicReference<>(pipeTaskInfo);
```

Comment:
Update pipeTaskInfoHolder to prevent undefined behavior in procedure after unlocking